packages/form/src/components/form/index.js

```diff
@@ -40,9 +40,7 @@ const Form = forwardRef( ( { children, name, onSubmit, ...props }, ref ) => {
 
 		startSubmit( name );
 
-		onSubmit( data )
-			.then( () => initForm( name, {} ) )
-			.finally( () => stopSubmit( name ) );
+		onSubmit( data ).finally( () => stopSubmit( name ) );
```

Comment:
Are we sure this won't cause the data for the previous page to submit when the next page is submitted?
For example, submit page 1, go to page 2, the request for page 2 will contain answers for both page 1 and page 2.

I'd recommend to give each page an appropriate suffix to the name, so we just select the correct set depending on what `currentPage` is.

Reply:
Thanks for catching that!
Already made the change to add a suffix to the form name based on the page index.